Provide CloudWatch log group with default forwarding to Splunk

JonRoma opened the pull request:
Provide CloudWatch log group with default log group subscription
rule forwarding log data to Splunk.

ddriddle commented on subscription-filter.tf:
```diff
@@ -0,0 +1,40 @@
+}
+
+resource "aws_lambda_permission" "default" {
+  count         = "${var.disable_splunk_subscription_filter == "true" ? 0 : 1}"
```
I don't think the == "true" is necessary.

ddriddle commented on subscription-filter.tf:
```diff
@@ -0,0 +1,40 @@
+}
+
+resource "aws_cloudwatch_log_subscription_filter" "default" {
+  count          = "${var.disable_splunk_subscription_filter == "true" ? 0 : 1}"
```
I don't think the == "true" is necessary.

ddriddle commented:
@JonRoma Please add a .drone.yml file:

```
pipeline:
  build:
    image: hashicorp/terraform:light
    commands:
      - terraform version
      - terraform validate -check-variables="false"
      - terraform fmt -check=true
```

ddriddle commented on subscription-filter.tf:
```diff
@@ -0,0 +1,40 @@
+
+# Get current region.
+
+data "aws_region" "current" {}
```
I would move this to the top of the file for better readability. Since it is referenced on line 3.

ddriddle commented on variables.tf:
```diff
@@ -0,0 +1,29 @@
+variable "disable_splunk_subscription_filter" {
```
This could be made more general. Change `disable_splunk_subscription_filter` to `disable_subscription_filter`

ddriddle commented on subscription-filter.tf:
```diff
@@ -0,0 +1,40 @@
+
+resource "aws_cloudwatch_log_subscription_filter" "default" {
+  count          = "${var.disable_splunk_subscription_filter == "true" ? 0 : 1}"
+  name           = "splunk-filter"
```
Change splunk-filter to filter.

ddriddle commented on variables.tf:
```diff
@@ -0,0 +1,29 @@
+variable "disable_splunk_subscription_filter" {
+  description = "Disable subscription filter to forward logs to Splunk"
```
Drop to Splunk.

JonRoma committed:
Provide CloudWatch log group with default forwarding to Splunk

Provide CloudWatch log group with default log group subscription
rule forwarding log data to Splunk.
